### PR TITLE
Add an End-to-End Test for PJS

### PR DESCRIPTION
**Pull request by FahadBSyed**

### CODE REVIEW

File: src/server/pjs_test.go [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+	}
+	for {
+		resp, err := listJobsClient.Recv()
+		if errors.Is(err, io.EOF) {
+			break
+		} else if err != nil {
+			t.Fatalf("unexpected error from list jobs client: %s", err.Error())
+		}
+		if resp.Job.Id != createJobResp.Job.Id {
+			t.Fatalf("job id doesn't match and we only created one job: expected: %d got: %d", createJobResp.Job.Id, resp.Job.Id)
```
Comment (Zhang-Muyang) [inline src/server/pjs_test.go:68]: (Based on a previous Jon's comment to my PR,) we should use `want` not `expected`

File: src/server/pjs_test.go [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+		CacheWrite: true,
+	})
+	if err != nil {
+		t.Fatalf("unexpected error from create job: %s", err.Error())
+	}
+	listJobsClient, err := c.PjsAPIClient.ListJob(ctx, &pjs.ListJobRequest{Job: createJobResp.Job})
+	if err != nil {
+		t.Fatalf("list jobs: %s", err.Error())
+	}
+	for {
```
Comment (Zhang-Muyang) [inline src/server/pjs_test.go:60]: it might be neater to use `grpc.Collect` to replace this for loop

File: src/server/pjs_test.go [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+	if err := listJobsClient.CloseSend(); err != nil {
+		t.Fatalf("unexpected error from list jobs client: close send: %s", err.Error())
+	}
+	listQueueClient, err := c.PjsAPIClient.ListQueue(ctx, &pjs.ListQueueRequest{})
+	if err != nil {
+		t.Fatalf("unexpected error from list queue client: %s", err.Error())
+	}
+	listQueueResp, err := listQueueClient.Recv()
+	if err != nil {
+		t.Fatalf("unexpected error from list queue client: recieve: %s", err.Error())
```
Comment (Zhang-Muyang) [inline src/server/pjs_test.go:80]: receive

**Review by Zhang-Muyang — APPROVED** [review]

File: src/internal/pjs/api_server.go [hunk 4]
```diff
@@ -562,7 +562,10 @@ func (a *apiServer) checkPermissions(ctx context.Context) error {
 	permissionResp, err := a.env.GetPermissionser.GetPermissions(ctx, &auth.GetPermissionsRequest{
 		Resource: &auth.Resource{Type: auth.ResourceType_CLUSTER},
 	})
-	if err != nil && !errors.Is(err, auth.ErrNotActivated) {
+	if err != nil {
+		if errors.Is(err, auth.ErrNotActivated) {
```
[the remark this answers is not in the document]
Comment (FahadBSyed) [reply src/internal/pjs/api_server.go:566]: Yeah sure, so if the error is auth.ErrNotActivated, then we proceed and try to access the response fields but whoops the response is nil and you get a nil pointer exception.

File: src/server/pjs_test.go (hunk 2 above)
Comment (FahadBSyed) [reply src/server/pjs_test.go:60]: I actually tried that and it didn't work, I think there's something that would have to be implemented for the response types which I can do, but I figured it wouldn't be worth it.

File: src/server/pjs_test.go [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+		t.Fatalf("unexpected error from create job: %v", err)
+	}
+	listJobsClient, err := c.PjsAPIClient.ListJob(ctx, &pjs.ListJobRequest{Job: createJobResp.Job})
+	if err != nil {
+		t.Fatalf("list jobs: %v", err)
+	}
+	for {
+		resp, err := listJobsClient.Recv()
+		if err != nil {
+			errors.I
```
Comment (Zhang-Muyang) [inline src/server/pjs_test.go:63]: typo

**Review by Zhang-Muyang — APPROVED** [review]

**Review by Zhang-Muyang — APPROVED** [review]

**Pull request merged**